### ISSUE TITLE
Uncaught TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by comikem** [mulesoft/api-workbench#281]
[Enter steps to reproduce below:]

1. Upgrade or Fresh install of Atom 1.9.0 on Windows or OS X
2. Select Packages -> API Workbench -> Create RAML Project 

**Atom Version**: 1.9.0
**System**: Mac OS X 10.11.6
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.35

### Stack Trace

Uncaught TypeError: Cannot read property '0' of undefined

```
At /Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2273

TypeError: Cannot read property '0' of undefined
    at editor.emitter.handlersByEventName.did-change (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2273:23)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:334:32
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at DisplayLayer.module.exports.DisplayLayer.bufferDidChange (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/display-layer.js:397:27)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.applyChange (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:711:27)
    at TextBuffer.module.exports.TextBuffer.revertToCheckpoint (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:896:16)
    at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:850:14)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:593:21)
    at TextBuffer.module.exports.TextBuffer.setText (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:533:19)
    at TextEditor.module.exports.TextEditor.setText (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:906:26)
    at AtomEditorElement.setText (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2284:20)
    at AtomEditorElement.handleDataChanged (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2290:14)
    at BasicComponent._bListener (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:110:23)
    at /Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:82:65
    at Array.forEach (native)
    at BasicBinding.set (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:82:28)
    at new TextField (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2699:31)
    at Object.texfField (/Users/mike/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:2789:12)
    at t.show (/Users/mike/.atom/packages/api-workbench/main.js:1:29437)
    at atom-workspace.u (/Users/mike/.atom/packages/api-workbench/main.js:1:23610)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (/Applications/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:1104:28)
    at EventEmitter.outerCallback (/Applications/Atom.app/Contents/Resources/app.asar/src/application-delegate.js:327:25)
    at emitTwo (events.js:100:13)
    at EventEmitter.emit (events.js:185:7)
```

### Commands

```
  2x -0:30.5.0 api-workbench:new-project (div.panels-item)
```

### Config

```json
{}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.35 (active)
linter, v1.11.14 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (active)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (active)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.0 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.0 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.38.1 (active)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.64.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.0 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.0 (active)
styleguide, v0.46.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.0 (active)
timecop, v0.33.1 (active)
tree-view, v0.208.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.32.2 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.36.2 (active)
language-gfm, v0.86.0 (active)
language-git, v0.13.0 (active)
language-go, v0.42.0 (active)
language-html, v0.44.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.22.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.0 (active)
language-less, v0.29.3 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.0 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.68.5 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.52.0 (active)
language-shellscript, v0.22.3 (active)
language-source, v0.9.0 (active)
language-sql, v0.21.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.8 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by geosmart** [comment]
the same error 

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.9.0
**System**: Unknown Windows Version
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.35


### Stack Trace

Uncaught TypeError: Cannot read property '0' of undefined

```
At C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2273

TypeError: Cannot read property '0' of undefined
    at editor.emitter.handlersByEventName.did-change (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2273:23)
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:125:28)
    at C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\text-editor.js:334:32
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:125:28)
    at DisplayLayer.module.exports.DisplayLayer.bufferDidChange (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\display-layer.js:397:27)
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.applyChange (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:711:27)
    at TextBuffer.module.exports.TextBuffer.revertToCheckpoint (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:896:16)
    at TextBuffer.module.exports.TextBuffer.transact (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:850:14)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:593:21)
    at TextBuffer.module.exports.TextBuffer.setText (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:533:19)
    at TextEditor.module.exports.TextEditor.setText (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\text-editor.js:906:26)
    at AtomEditorElement.setText (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2284:20)
    at AtomEditorElement.handleDataChanged (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2290:14)
    at BasicComponent._bListener (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:110:23)
    at C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:82:65
    at Array.forEach (native)
    at BasicBinding.set (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:82:28)
    at new TextField (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2699:31)
    at Object.texfField (C:\Users\Think\.atom\packages\api-workbench\node_modules\atom-ui-lib\dist\UI.js:2789:12)
    at t.show (C:\Users\Think\.atom\packages\api-workbench\main.js:1:29437)
    at atom-workspace.u (C:\Users\Think\.atom\packages\api-workbench\main.js:1:23610)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\atom-environment.js:1104:28)
    at EventEmitter.outerCallback (C:\Users\Think\AppData\Local\atom\app-1.9.0\resources\app.asar\src\application-delegate.js:327:25)
    at emitTwo (events.js:100:13)
    at EventEmitter.emit (events.js:185:7)
```

### Commands

```
     -7:20.6.0 api-workbench:new-project (atom-pane.pane.active)
     -0:50.7.0 api-workbench:popular-apis (a.btn-issue.btn.btn-error)
     -0:31.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  2x -0:15.5.0 api-workbench:editor-tools (atom-text-editor.editor.mini.is-focused)
     -0:03.8.0 api-workbench:new-project (atom-pane.pane.active)
```

### Config

```json
{}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.35 (active)
linter, v1.11.14 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (active)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (active)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.0 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.0 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.38.1 (active)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.64.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.0 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.0 (active)
styleguide, v0.46.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.0 (active)
timecop, v0.33.1 (active)
tree-view, v0.208.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.32.2 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.36.2 (active)
language-gfm, v0.86.0 (active)
language-git, v0.13.0 (active)
language-go, v0.42.0 (active)
language-html, v0.44.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.22.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.0 (active)
language-less, v0.29.3 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.0 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.68.5 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.52.0 (active)
language-shellscript, v0.22.3 (active)
language-source, v0.9.0 (active)
language-sql, v0.21.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.8 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```


```

**Comment by moander** [comment]
Same here..

http://pastebin.com/raw/n0YwihVE

**Comment by lenovomadp** [comment]
same error. can any one solve it?

**Comment by ddenisenko** [comment]
Yes, it looks like Atom update brought us something breaking this. We'll try to restore it fast.

**Comment by dreamflyer** [comment]
fixed in https://github.com/mulesoft-labs/ui-libs/pull/3

**Comment by ddenisenko** [comment]
That is because the fix is not published yet. It will be published in version 0.8.36 later today.

**Comment by ddenisenko** [comment]
The fix is published in 0.8.36. Please restart Atom after update.

**Comment by lenovomadp** [comment]
I updated and confirmed it was fixed.  Great! Thank you!


发件人: Denis Denisenko [mailto:notifications@github.com]
发送时间: 2016年8月5日 14:27
收件人: mulesoft/api-workbench <api-workbench@noreply.github.com>
抄送: Xiaolei XL7 Chen <chenxl7@lenovo.com>; Comment <comment@noreply.github.com>
主题: Re: [mulesoft/api-workbench] Uncaught TypeError: Cannot read property '0' of undefined (#281)


The fix is published in 0.8.36. Please restart Atom after update.
